WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed. 
*****************************************
WARNING: No training data specified
WARNING: No training data specified
using world size: 8 and model-parallel size: 8 
> padded vocab (size: 150528) with 0 dummy tokens (new size: 150528)
WARNING: No training data specified
WARNING: No training data specified
WARNING: No training data specified
WARNING: No training data specified
WARNING: No training data specified
WARNING: No training data specified
> initializing model parallel with size 8
> Loading task configs
    Task StereoSet loaded from config tasks/ethnic/StereoSet.yaml
> Successfully load 1 task
> Set tokenizer as a icetk-glm-130B tokenizer! Now you can get_tokenizer() everywhere.
global rank 3 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_03_model_states.pt
global rank 6 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_06_model_states.pt
global rank 0 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_00_model_states.pt
global rank 1 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_01_model_states.pt
global rank 5 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_05_model_states.pt
global rank 7 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_07_model_states.pt
global rank 2 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_02_model_states.pt
global rank 4 is loading checkpoint /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_04_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_06_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_07_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_00_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_03_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_04_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_01_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_05_model_states.pt
  successfully loaded /mnt/yrfs/aohan/checkpoints/glm-130b-sat/49300/mp_rank_02_model_states.pt
> Checkpoint loaded in 23.8s
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
Traceback (most recent call last):
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
Traceback (most recent call last):
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 67, in <module>
    main()
      File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
main()    main()

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
        main()main()    main()

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
    main()
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
                evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)

evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks
      File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks
evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks
        evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)main()
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 62, in main
    evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks
        task = task_class(model, tokenizer, config)task = task_class(model, tokenizer, config)        
        
task = task_class(model, tokenizer, config)task = task_class(model, tokenizer, config)  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__
task = task_class(model, tokenizer, config)

task = task_class(model, tokenizer, config)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__
    evaluate_all_tasks(args.data_path, model, tokenizer, args.task, task_classes)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluate.py", line 35, in evaluate_all_tasks
    task = task_class(model, tokenizer, config)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__
    super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
    super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
    task = task_class(model, tokenizer, config)        
super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 218, in __init__


  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
      File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
    super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
    super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    super().__init__(ModelForEvaluation, _IceTokenizer, MultiChoiceTaskConfig)
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 41, in __init__
    self.config.metrics = list(self.metrics.keys())
  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}
    return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}
AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
    return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}
AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
    return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}
AttributeError    : return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}type object 'MultiChoiceTaskConfig' has no attribute 'metrics'

AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
        return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}self.config.metrics = list(self.metrics.keys())

  File "/mnt/yrfs/xuyifan/GLM-130B/evaluation/tasks.py", line 35, in metrics
    return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}
AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
    return {metric: DEFAULT_METRICS[metric] for metric in self.config.metrics}
AttributeError: type object 'MultiChoiceTaskConfig' has no attribute 'metrics'
ERROR:torch.distributed.elastic.multiprocessing.api:failed (exitcode: 1) local_rank: 0 (pid: 422003) of binary: /mnt/yrfs/qinkai/miniconda3/envs/largescale/bin/python
Traceback (most recent call last):
  File "/mnt/yrfs/qinkai/miniconda3/envs/largescale/bin/torchrun", line 33, in <module>
    sys.exit(load_entry_point('torch==1.12.0', 'console_scripts', 'torchrun')())
  File "/mnt/yrfs/qinkai/miniconda3/envs/largescale/lib/python3.9/site-packages/torch/distributed/elastic/multiprocessing/errors/__init__.py", line 345, in wrapper
    return f(*args, **kwargs)
  File "/mnt/yrfs/qinkai/miniconda3/envs/largescale/lib/python3.9/site-packages/torch/distributed/run.py", line 761, in main
    run(args)
  File "/mnt/yrfs/qinkai/miniconda3/envs/largescale/lib/python3.9/site-packages/torch/distributed/run.py", line 752, in run
    elastic_launch(
  File "/mnt/yrfs/qinkai/miniconda3/envs/largescale/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 131, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/mnt/yrfs/qinkai/miniconda3/envs/largescale/lib/python3.9/site-packages/torch/distributed/launcher/api.py", line 245, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
/mnt/yrfs/xuyifan/GLM-130B/evaluate.py FAILED
------------------------------------------------------------
Failures:
[1]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 1 (local_rank: 1)
  exitcode  : 1 (pid: 422004)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[2]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 2 (local_rank: 2)
  exitcode  : 1 (pid: 422005)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[3]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 3 (local_rank: 3)
  exitcode  : 1 (pid: 422006)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[4]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 4 (local_rank: 4)
  exitcode  : 1 (pid: 422007)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[5]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 5 (local_rank: 5)
  exitcode  : 1 (pid: 422009)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[6]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 6 (local_rank: 6)
  exitcode  : 1 (pid: 422010)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
[7]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 7 (local_rank: 7)
  exitcode  : 1 (pid: 422011)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2022-09-24_21:02:36
  host      : node14
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 422003)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================
